app/lib/widgets/wallets/add_wallet.dart

```diff
@@ -114,6 +114,10 @@ class _NewWalletState extends State<NewWallet> {
       return false;
     }
 
+    if (!walletSecret.startsWith('0x') && walletSecret.length != 56) {
+      secretError = 'Invalid seed length';
+      return false;
+    }
```

Comment:
this would break the validation for the other types like mnemonic

Reply:
I added a new condition that checks if the stellar secret (which always starts with 'S') has a length of 56.